encodings/fastlanes/src/bitpacking/compress.rs

```diff
@@ -324,35 +381,28 @@ where
     let full_chunks_range =
         (first_chunk_is_sliced as usize)..(num_chunks - last_chunk_is_sliced as usize);
 
+    // Index into the builder's uninit values buffer.
+    let mut out_idx = 0;
     if first_chunk_is_sliced {
         let chunk = &packed[..elems_per_chunk];
         let mut decoded = [UnsignedT::zero(); 1024];
         // SAFETY:
         // 1. chunk is elems_per_chunk.
         // 2. decoded is exactly 1024.
         unsafe { BitPacking::unchecked_unpack(bit_width, chunk, &mut decoded) };
-        builder
-            .values
-            .extend_from_slice(transmute(&decoded[offset..]));
+        uninit.copy_from_init(out_idx, 1024 - offset, transmute(&decoded[offset..]));
```

Comment:
The `uninit` isn't stateful, the whole point is that you have access to the whole range while the `uninit` is live. The downside is we need to keep track of the `out_idx` ourself